### PR TITLE
Task/adq 40 remove referencerange observation

### PR DESCRIPTION
**Pull request by jhulbert-va**
[ADQ-40](https://vasdvp.atlassian.net/browse/ADQ-40)

* Reference Range is removed if category is 'vital-signs'
* Added Junit to test functionality 

**The db patient is only returning a case with vital-signs so the other case has not been tested. Except by the junit tests.**

### CODE REVIEW

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/ObservationTransformer.java [hunk 1]
```diff
@@ -64,7 +65,10 @@ public Observation apply(CdwObservation cdw) {
         .valueCodeableConcept(valueCodeableConcept(cdw.getValueCodeableConcept()))
         .interpretation(interpretation(cdw.getInterpretation()))
         .comments(cdw.getComments())
-        .referenceRange(referenceRanges(cdw.getReferenceRanges()))
+        .referenceRange(
+            (cdw.getCategory().getCoding().get(0).getCode().value().equals(CdwObservationCategoryCode.VITAL_SIGNS.value()))
```
Comment (eclendenning-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/ObservationTransformer.java]: Possible NPE. Consider extracting into a helper method that checks that each piece of the chain is present. Check Transformers.java for useful utility methods.

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/ObservationTransformer.java [hunk 2]
```diff
@@ -49,6 +49,7 @@ public Observation apply(CdwObservation cdw) {
      * Specimen reference is omitted since we do not support the a specimen resource and
      * do not want dead links
      */
+    String categoryCode = cdw.getCategory().getCoding().get(0).getCode().value();
```
Comment (bschofield-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/ObservationTransformer.java]: `.get(0)` without checking size can blow up

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/ObservationTransformer.java [hunk 3]
```diff
@@ -64,7 +65,11 @@ public Observation apply(CdwObservation cdw) {
         .valueCodeableConcept(valueCodeableConcept(cdw.getValueCodeableConcept()))
         .interpretation(interpretation(cdw.getInterpretation()))
         .comments(cdw.getComments())
-        .referenceRange(referenceRanges(cdw.getReferenceRanges()))
+        .referenceRange(
```
Comment (bschofield-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/ObservationTransformer.java]: Let's move this logic into to `referenceRanges(...)` and update the signature to be `referenceRanges(cdw.getCategory(), cdw.getReferenceRanges())`

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/ObservationTransformer.java [hunk 4]
```diff
@@ -296,7 +291,18 @@ SimpleQuantity referenceRangeQuantity(CdwObservationRefRangeQuantity maybeCdw) {
         .build();
   }
 
-  List<ObservationReferenceRange> referenceRanges(CdwReferenceRanges maybeCdw) {
+  List<ObservationReferenceRange> referenceRanges(
+      CdwReferenceRanges maybeCdw, CdwCategory cdwCategory) {
+    List<CdwCategory.CdwCoding> categoryCoding = cdwCategory.getCoding();
+    if (categoryCoding.size() > 0
+        && !StringUtils.isEmpty(categoryCoding.get(0).getCode().value())
```
Comment (eclendenning-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/observation/ObservationTransformer.java]: If getCode() is null, .value() will throw an NPE.

(one more check!)
(one more check!)

**Pull request merged**